tests/api/runtimes/base.py

```diff
@@ -301,7 +308,10 @@ def _assert_pod_create_called(
         expected_env["MLRUN_NAMESPACE"] = self.namespace
         self._assert_pod_env(pod_env, expected_env)
         for env_variable in pod_env:
-            if env_variable["name"] == "MLRUN_EXEC_CONFIG":
+            if (
+                not isinstance(env_variable, V1EnvVar)
```

Comment:
Why are you limiting yourself like this? Again here I would use the method mentioned above and convert. This way it doesn't matter what type is within the structure.